### PR TITLE
Initial commit for the Gizmos solution and Doodads project files.

### PR DESCRIPTION
**Pull request by DogooFalchion**
This commit creates the /firework command in the Doodads plugin.

Doodad - a trivial ornament or gadget, especially one whose name the speaker does not know or cannot recall.
"the latest electronic doodads"

### CODE REVIEW

**Review by AxisKriel — APPROVED** [review]
Looks good :+1:

I like the `ICommandHandler` + command classes, definitely more structured than the half-baked command framework it is based upon (aka TShock's).

File: Doodads/DoodadsPlugin.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+using Terraria;
+using TerrariaApi.Server;
+using TShockAPI;
+
+namespace Doodads
+{
+	/// <summary>
+	/// The main plugin class should always be decorated with an ApiVersion attribute. The current API Version is 1.22
+	/// </summary>
+	[ApiVersion(1, 22)]
```
Comment (AxisKriel) [inline Doodads/DoodadsPlugin.cs]: Not something that needs changing in this PR seeing as this is an initial commit, but the current API version is 1.25.

Comment (hakusaro) [reply Doodads/DoodadsPlugin.cs]: No, this should be 1.25...

**Review by hakusaro — CHANGES_REQUESTED** [review]
All good side the API version.

**Comment by DogooFalchion** [comment]
Oh, ok.  I was going based off the template plugin.  Might want to get that updated :)

**Comment by AxisKriel** [comment]
Our resources are so damn outdated that you really shouldn't base your plugins off them. Not without a check to bring everything up to date, at least.

**Comment by hakusaro** [comment]
@Enerdy that's our fault 💩 .

**Review by hakusaro — APPROVED** [review]

**Comment by DogooFalchion** [comment]
Especially when they are linked on the readme 👎

**Pull request merged**